### DISCUSSION TITLE
Moving target / Still target.

### DISCUSSION THREAD

**Discussion by gohlas** [TillFleisch/ESPHome-HLK-LD2450#31] [General]
Hi,
Thanks for the great component.
I'm wondering where this component supports only moving, or are they showing together in the same.
I noticed that if I just stand still the sensor release occupancy.
Just compared with the other PR and I see that there has both as a separate entities.

I prefer rather this component, since there is the polygon solution that works better for me.

**Comment by gohlas** [comment]
Just found on this post that still detection on LD2450 is not that good.
https://community.home-assistant.io/t/ld2410c-vs-2410b-vs-ld2410s-vs-2411-vs-hlk-ld2420-vs-ld2450/652599

Sadly!

**Comment by TillFleisch** [comment]
>  I noticed that if I just stand still the sensor release occupancy.

This is an inherent problem with mm-Wave Sensors. Their "still-detection" capabilities rely on the detection of very small movements such as breathing. Under the right conditions, the sensor may not see these small movements and therefore show "unoccupied".
Moving closer to the sensor may help resolve this issue.

From a software standpoint (what this external component can accomplish), there really is not a lot that can be done to circumvent this issue. You can try increasing timeouts, such that the final output is less tolerant to intermittent falsely detected user-absence. I found that `target_timeout: 10s` works well for my use case. A higher value may work even better and if I recall correctly my light automation allows for 5 minutes of absence before turning of the lights.

If you want to differentiate between still and moving targets, the `speed` property may be used in combination with a template sensor.
